src/main/java/rx/internal/schedulers/SchedulerContextException.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright 2016 Netflix, Inc.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package rx.internal.schedulers;
+
+/**
+ * Used only for providing context around where work was scheduled should an error occur in a different thread.
+ */
+public class SchedulerContextException extends Exception {
+    /**
+     * Constant to use when disabled
+     */
+    private static final Throwable CONTEXT_MISSING = new SchedulerContextException("Missing context. Enable by "//
+            + "setting the system property \"rxjava.captureSchedulerContext=true\" at startup or "
+            + "calling \"rx.internal.schedulers.SchedulerContextException.setCaptureSchedulerContext(true)\" at runtime");
+    /**
+     * cache of the system property
+     */
+    private static volatile boolean captureSchedulerContext;
+
+    static {
+        CONTEXT_MISSING.setStackTrace(new StackTraceElement[0]);
+
+        try {
+            captureSchedulerContext = System.getProperty("rxjava.captureSchedulerContext", "false").equals("true");
+        } catch (Exception e) {
+            captureSchedulerContext = false;
+        }
+    }
+
+    /**
+     * Immediately change the behavior to of the {@link SchedulerContextException#create()} method.
+     * @param captureSchedulerContext
+     */
+    public static void setCaptureSchedulerContext(boolean captureSchedulerContext) {
```

Comment:
It wasn't an API until it was changed to `volatile boolean` and was just always reading from `System.getProperty()`.